### DISCUSSION TITLE
Eclipse p3com

### DISCUSSION THREAD

**Discussion by JakubSosnovec** [eclipse-iceoryx/iceoryx#1961] [Show and tell]
Hi iceoryx developers!

I would like to share with you that we have just recently pushed the initial contribution of a new Eclipse project -- "Eclipse p3com" here: https://github.com/eclipse-p3com/p3com

Eclipse p3com is coming from the automotive group in NXP Semiconductors, but it has now been contributed as an open source project under the Eclipse foundation, in particular as part of the Eclipse Software Defined Vehicles working group.

Eclipse p3com is basically nothing else than infrastructure for writing gateways for Eclipse iceoryx, a little bit like the existing DDS gateway in iceoryx, but much more dynamic and general. It is fully an extension of iceoryx, it is just using the public API and various introspection and discovery mechanisms. I think it can be interesting for you guys.

For now, out of the box, the project is not very much useful for anything -- it only has network transport layers, which are mostly intended for debug purposes, because if you wanted to do networking-based communication, you can much more easily used some proper middlewares such as DDS or MQTT. But gradually, we intend to also contribute our NXP-platform-specific PCIe transport layer which should nicely demonstrate the full capabilities of p3com. Also, we need to write a guide on how to implement new transport layers, because thats the whole idea -- that p3com is just the infrastructure which should enable developers to easily implement transport layers to leverage performance of their own HW platforms.

Would you guys be interested in having a call to discuss Eclipse p3com and have a sync?

Also, we are planning to soon open a PR with iceoryx to merge our FreeRTOS platform implementation for iceoryx. Our lawyers are still taking their time to approve this, but hopefully that will happen soon.

Please, share any thoughts, feedback or questions that you have.

**Comment by elfenpiff** [comment]
@JakubSosnovec I think this is awesome and I would love to sync here. Here we could also share where the journey with iceoryx goes and what our next big features will be.

The FreeRTOS would complete the iceoryx platforms nicely, with it the only platform we have to tackle soon would be android.
